src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,107 @@
+import { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+import { Person } from '../types/Person';
+
+interface Props {
+  people: Person[]
+  delay: number
+  onSelected: (person: Person | null) => void
+}
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  delay,
+  onSelected,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isListShown, setIsListShown] = useState(false);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(
+    debounce((value: string) => {
+      setAppliedQuery(value);
+      setIsListShown(true);
+    }, delay),
+    [],
+  );
+
+  const filteredPeople = useMemo(() => {
+    const normalizedValue = appliedQuery.trim().toLowerCase();
+
+    return people
+      .filter(person => person.name.toLowerCase().includes(normalizedValue));
+  }, [appliedQuery, people]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value.trim().toLowerCase();
+
+    setIsListShown(false);
+    applyQuery(inputValue);
+
+    setQuery(event.target.value);
+  };
+
+  const handleInputSelect = (selectedPerson: Person) => {
+    setQuery(selectedPerson.name);
+    onSelected(selectedPerson);
+    setIsListShown(false);
+  };
+
+  return (
+    <div className={cn(`
+      dropdown
+      ${isListShown ? 'is-active' : ''}`)}
```

Comment:
Why are you specifying a class via a ternary operator?
Use classnames library for add classes with condition